clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -890,6 +893,10 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                     if (topic == null || topic.trim().isEmpty())
                         throw new IllegalArgumentException("Topic collection to subscribe to cannot contain null or empty topic");
                 }
+
+                if (assignors.isEmpty())
+                    throw new IllegalArgumentException("Must specify at least one partition assigner");
+
```

Comment:
Do we need to use ConfigException here?

Reply:
Yeah, it's a good question. `IllegalArgumentException` doesn't feel quite right. Another option would be `IllegalStateException`. Also, we should probably mention the configuration property in the exception message.

Reply:
@hachikuji  Updated the PR. Please check.